core/src/layers/throttle.rs

```diff
@@ -0,0 +1,332 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+use std::io::SeekFrom;
+use std::num::NonZeroU32;
+use std::sync::Arc;
+use std::task::{Context, Poll};
+
+use async_trait::async_trait;
+use bytes::Bytes;
+use chrono::Duration;
+use governor::clock::{Clock, DefaultClock, QuantaInstant};
+use governor::middleware::{RateLimitingMiddleware, StateInformationMiddleware, StateSnapshot};
+use governor::state::{InMemoryState, NotKeyed};
+use governor::{NegativeMultiDecision, NotUntil, Quota, RateLimiter, RatelimitedStream};
+
+use crate::raw::*;
+use crate::*;
+
+/// ThrottleLayer can help users to control the max bandwidth that used by OpenDAL.
+#[derive(Clone)]
+pub struct ThrottleLayer {
+    // TODO: NonZeroU32 or u32, since user enter 0 will cause panic
+    max_burst: u32,
+}
+
+impl ThrottleLayer {
+    /// Create a new ThrottleLayer will specify quota
+    pub fn new(max_burst: u32) -> Self {
```

Comment:
What's the unit of `max_burst`?

Reply:
It's Bytes (e.g. 9kiB/s should be `9*1024`)

Reply:
Please make it clear in comments.

Reply:
I think you could just name the `max_burst`, to `max_burst_byte`

Reply:
> I think you could just name the `max_burst`, to `max_burst_byte`

Thanks @Ji-Xinyou, that's a good suggestion! I’ll rename to `max_burst_byte`. (Although I originally named it max_burst to align with the [Quota source code](https://github.com/antifuchs/governor/blob/9991cc31510ba7e57bfb47dc18522b9404001fd6/governor/src/quota.rs#LL64C38-L64C38)